src/modules/m_sasl.cpp

```diff
@@ -35,6 +35,10 @@ static void SendSASL(const parameterlist& params)
 {
 	if (!ServerInstance->PI->SendEncapsulatedData(params))
 	{
+		User* u = ServerInstance->FindUUID(params[2]);
+		if (u)
+			u->WriteNumeric(904, "%s :SASL authentication failed", u->nick.c_str());
+
```

Comment:
Have you checked whether `authExt` needs to be unset here?

Reply:
As far as I can tell it's unset in the OnUserConnect catch by checking if the user has an active SaslAuthenticator, then aborting and unsetting it. I have not attempted a deeper "debug" level check of this, but clients like (iirc)Weechat do a SASL timeout and continue connecting if they don't get a registration timeout, which would have the same effect on the set SaslAuthenticator.

Reply:
I placed a simple debug snomask message inside the if block at line 346 (if user has a SaslAuthenticator) and it is catching the continued connect after attempting a SASL auth, therefore aborting and unsetting it.